pkg/controller/core/queuedworkload_controller.go

```diff
@@ -58,7 +63,21 @@ func NewQueuedWorkloadReconciler(queues *queue.Manager, cache *cache.Cache) *Que
 //+kubebuilder:rbac:groups=kueue.x-k8s.io,resources=queuedworkloads/finalizers,verbs=update
 
 func (r *QueuedWorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	// No-op. All work is done in the event handler.
+	var wl kueue.QueuedWorkload
+	if err := r.client.Get(ctx, req.NamespacedName, &wl); err != nil {
+		// we'll ignore not-found errors, since there is nothing to do.
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	log := ctrl.LoggerFrom(ctx).WithValues("queuedWorkload", klog.KObj(&wl))
+	ctx = ctrl.LoggerInto(ctx, log)
+	log.V(2).Info("Reconciling QueuedWorkload")
+
+	status := workloadStatus(&wl)
+	if status == pending && !r.queues.QueueExists(&wl) {
+		err := workload.UpdateWorkloadStatus(ctx, r.client, &wl, kueue.QueuedWorkloadAdmitted, corev1.ConditionFalse,
```

Comment:
if the controller restarts, all workloads are reconciled. In that case, you might be updating all the statuses again, even though the reason didn't change.

Perhaps you need `UpdateWorkloadStatusIfChanged`. You still need `UpdateWorkloadStatus` to be used by the scheduler, as we want to know about the last scheduling attempt anyways.

Reply:
done